src/js/protocols/CapacitorSerial.js

```diff
@@ -295,6 +319,21 @@ class CapacitorSerial extends EventTarget {
             return new Uint8Array(0);
         }
 
+        // Strip 0x prefix
+        if (hexString.startsWith("0x") || hexString.startsWith("0X")) {
+            hexString = hexString.slice(2);
+        }
+
+        // Validate characters
+        if (!/^[0-9a-fA-F]+$/.test(hexString)) {
+            throw new Error(`Invalid hex string: contains non-hex characters`);
+        }
+
+        // Validate length
+        if (hexString.length % 2 !== 0) {
+            throw new Error(`Invalid hex string: odd length (${hexString.length})`);
+        }
```

Comment:
Actually we might check this - as it would indicate corruption - but would like this more centralized.

There is a lot of duplicate code in protocols as we are using serial facade to link in other protocols with their own implementations. Added some thoughts for improvement in https://github.com/betaflight/betaflight-configurator/issues/4722.